components/gallery/List.vue

```diff
@@ -0,0 +1,303 @@
+                :caption="caption"
+                :class="imageClasses"
+                :image="image"
+                mode="fullbleed"
+                @error-video="videoError"
+                @error-image="imageError"
+                @loaded-image="imageLoaded"
+                @loaded-video="videoLoaded"
+            />
+        </transition>
+    </div>
+</template>
+<script>
+
+import _dropRight from "lodash/dropRight"
+import _takeRight from "lodash/takeRight"
+
+export default {
+    name: "List",
+    props: {
+        items: {
+            type: Array,
+            default: () => []
+        },
+    },
+    data() {
+        return {
+            isImageLoaded: false,
+            isVideoLoaded: false,
+            isImageError: false,
+            isVideoError: false,
+            caption: '',
+            image: null
+        }
+    },
+    computed: {
+        imageClasses() {
+            return [
+                {
+                    'has-loaded-media': this.isImageLoaded || this.isVideoLoaded,
+                    'has-full-media-error': this.isVideoError && this.isImageError && !this.isImageLoaded && !this.isVideoLoaded,
+                }
+            ]
+
+        },
+        preparedList() {
+            let tempData = this.items
+            let {right, left} = this.shapeListData(tempData, 3)
+            let trimmed = this.shapeListData(left, 2)
+
+            return [
+                [
+                    {
+                        listItems: trimmed.left
+                    },
+                ],
+                [
+                    {
+                        listItems: trimmed.right
+                    },
+                    {
+                        listItems: right
+                    }
+                ]
+
+            ]
+        },
+    },
+    methods: {
+        videoLoaded() {
+            this.isVideoLoaded = true
+        },
+        imageLoaded() {
+            this.isImageLoaded = true
+        },
+        videoError() {
+            this.isVideoError = true
+        },
+        imageError() {
+            this.isImageError = true
+        },
+
+        mouseOver(item) {
+            this.caption = item.featuredImage.node.caption !== null ? item.featuredImage.node.caption : `Who the fuxt is ${item.title}?`
+            this.image = item.featuredImage.node
+        },
+        mouseOut() {
+            this.isVideoLoaded = false
+            this.isImageLoaded = false
+            this.isVideoError = false
+            this.isImageError = false
+            this.caption = ''
+            this.image = null
+        },
+        shapeListData(array, offset) {
+            let count = array.length
+            let trimLastCount = Math.floor(count / offset)
+            let rightArray = _takeRight(array, trimLastCount)
+            let leftArray = _dropRight(array, trimLastCount)
+
+            return {
+                left: leftArray,
+                right: rightArray
+            }
+        },
```

Comment:
If you are going to do a function like this, then you can actually do this outside of methods. If the method has no need for anything Vue related, then make a utility function instead and call that.

Reply:
Good call and moved to `utils`